xds/internal/xdsclient/bootstrap/bootstrap_test.go

```diff
@@ -1034,4 +1038,23 @@ func TestCredsBuilders(t *testing.T) {
 	if got, want := i.Name(), "insecure"; got != want {
 		t.Errorf("insecureCredsBuilder.Name = %v, want %v", got, want)
 	}
+
+	tcb := &tlsCredsBuilder{}
+	if _, err := tcb.Build(nil); err == nil {
+		t.Errorf("tlsCredsBuilder.Build succeeded, want failure")
+	}
```

Comment:
This is expected to fail because the tls creds plugin does not like an `nil` config, is that right? Is this the behavior in other languages as well? Would it be more appropriate to treat a `nil` config like an empty config and use system-wide certs?

Reply:
Good catch, I think we want to accept `config` not being set. I verified that c-core accepts it as valid and treats it like empty by running the C++ xds example with a tls cred type but no config.

I fixed the code by handling nil config in the bundle and changed this test to do the same as for insecure/google, which support no config.